docs/storage_snippets.py

```diff
@@ -0,0 +1,257 @@
+"""
+
+import time
+
+from google.cloud import storage
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+@snippet
+def storage_get_started(client, to_delete):
+    # [START storage_get_started]
+    client = storage.Client()
+    bucket = client.get_bucket('bucket-id-here')
+    # Then do other things...
+    blob = bucket.get_blob('/remote/path/to/file.txt')
+    print(blob.download_as_string())
+    blob.upload_from_string('New contents!')
+    blob2 = bucket.blob('/remote/path/storage.txt')
+    blob2.upload_from_filename(filename='/local/path.txt')
+    # [END storage_get_started]
+
+    to_delete.append(bucket)
+
+
+@snippet
+def client_bucket_acl(client, to_delete):
+    bucket_name = 'system-test-bucket'
+    bucket = client.bucket(bucket_name)
+    bucket.create()
+
+    # [START client_bucket_acl]
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+    acl = bucket.acl
+    # [END client_bucket_acl]
+    to_delete.append(bucket)
+
+    # [START acl_user_settings]
+    acl.user('me@example.org').grant_read()
+    acl.all_authenticated().grant_write()
+    # [END acl_user_settings]
+
+    # [START acl_save]
+    acl.save()
+    # [END acl_save]
+
+    # [START acl_revoke_write]
+    acl.all().grant_read().revoke_write()
+    # [END acl_revoke_write]
+
+    # [START acl_save_bucket]
+    bucket.acl.save(acl=acl)
+    # [END acl_save_bucket]
+
+    # [START acl_print]
+    print(list(acl))
+    # [{'role': 'OWNER', 'entity': 'allUsers'}, ...]
+    # [END acl_print]
+
+
+@snippet
+def download_to_file(client, to_delete):
+    # [START download_to_file]
+    from google.cloud.storage import Blob
+
+    client = storage.Client(project='my-project')
+    bucket = client.get_bucket('my-bucket')
+    encryption_key = 'aa426195405adee2c8081bb9e7e74b19'
+    blob = Blob('secure-data', bucket, encryption_key=encryption_key)
+    with open('/tmp/my-secure-file', 'wb') as file_obj:
+        blob.download_to_file(file_obj)
+    # [END download_to_file]
+
+    to_delete.append(blob)
+
+
+@snippet
+def upload_from_file(client, to_delete):
+    # [START upload_from_file]
+    from google.cloud.storage import Blob
+
+    client = storage.Client(project='my-project')
+    bucket = client.get_bucket('my-bucket')
+    encryption_key = 'aa426195405adee2c8081bb9e7e74b19'
+    blob = Blob('secure-data', bucket, encryption_key=encryption_key)
+    with open('my-file', 'rb') as my_file:
+        blob.upload_from_file(my_file)
+    # [END upload_from_file]
+
+    to_delete.append(blob)
+
+
+@snippet
+def get_blob(client, to_delete):
+    # [START get_blob]
+    client = storage.Client()
+    bucket = client.get_bucket('my-bucket')
+    print(bucket.get_blob('/path/to/blob.txt'))
+    # <Blob: my-bucket, /path/to/blob.txt>
+    print(bucket.get_blob('/does-not-exist.txt'))
+    # None
```

Comment:
@daspecster In the other snippets files, @tseaver has used an assert where a print statement once was, e.g.

```
assert bucket.get_blob('/does-not-exist.txt') is None
```

Though through the ongoing discussion in #212, hopefully we can test actual interpreter snippets.